src/main/java/GUI/RoleRelatedViews/Referee/RefereeControls.java

```diff
@@ -0,0 +1,73 @@
+package GUI.RoleRelatedViews.Referee;
+
+import GUI.FootballMain;
+import Service.MainController;
+import com.vaadin.flow.component.UI;
+import com.vaadin.flow.component.button.Button;
+import com.vaadin.flow.component.orderedlayout.VerticalLayout;
+import com.vaadin.flow.router.PageTitle;
+import com.vaadin.flow.router.Route;
+import com.vaadin.flow.server.VaadinService;
+import com.vaadin.flow.server.VaadinSession;
+
+@Route(value = "REF", layout = FootballMain.class)
+@PageTitle("Referee Controls")
+public class RefereeControls extends VerticalLayout {
+    public static final String VIEW_NAME = "Referee Controls";
+
+    public RefereeControls() {
+        setSizeFull();
+        setJustifyContentMode(JustifyContentMode.START);
+        setAlignItems(Alignment.START);
+        buildUI();
+    }
+
+    private void buildUI() {
+        createShowActiveGameButton();
+        createDisplayScheduleGame();
+    }
+
+    private void createDisplayScheduleGame() {
+        String buttonTaskName = "View Game Schedule";
+        Button watchGamesBtn = new Button(buttonTaskName);
+        watchGamesBtn.addClickListener(e -> {
+
+            UI lastUI = UI.getCurrent();
+            VaadinSession se = VaadinSession.getCurrent();
+            String username = getUsernameFromSession();
+
+                UI.setCurrent(lastUI);
+                VaadinSession.setCurrent(se);
+                MainController.DisplayScheduledGame(username);
+
```

Comment:
where is the Thread start part? did you test this button?

Reply:
He doesn't need a thread for displaying games